mage_ai/frontend/components/ContextMenu/index.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState, useCallback } from 'react';
+
+import BlockType from '@interfaces/BlockType';
+import FlyoutMenu from '@oracle/components/FlyoutMenu';
+import styled from 'styled-components';
+import { FileContextEnum } from '@components/FileBrowser';
+import { UNIT } from '@oracle/styles/units/spacing';
+import { HEADER_HEIGHT, HEADER_Z_INDEX } from '@components/constants';
+
+export type ContextMenuSharedProps = {
+  deleteBlockFile: (b: BlockType) => void;
+};
+
+export type ContextMenuProps = {
+  children: any;
+  areaRef: any;
+  type: ContextMenuEnum;
+  useContextItem: boolean;
+} & ContextMenuSharedProps;
+
+export enum ContextMenuEnum {
+  FILE_BROWSER = 'file_browser',
+}
+
+export interface ContextAreaProps {
+  setContextItem?: (item: any) => void;
+}
+
+export interface ContextItemType {
+  data: any;
+  type: any;
+}
+
+const ContainerStyle = styled.div<{
+  top?: number;
+  left?: number;
+  right?: number;
+  bottom?: number;
+  width?: number;
+}>`
+  position: fixed;
+  z-index: ${HEADER_Z_INDEX + 100};
+
+  ${props => typeof props.left !== 'undefined' && `
+    left: ${props.left + 5}px;
+  `}
+
+  ${props => typeof props.right !== 'undefined' && `
+    right: ${props.right}px;
+  `}
+
+  ${props => typeof props.top !== 'undefined' && `
+    top: ${props.top}px;
+  `}
+
+  ${props => typeof props.bottom !== 'undefined' && `
+    bottom: ${props.bottom}px;
+  `}
+
+  ${props => typeof props.top === 'undefined' && typeof props.bottom === 'undefined' && `
+    top: ${HEADER_HEIGHT - (UNIT * 1)}px;
+  `}
+
+  ${props => props.width && `
+    width: ${props.width}px;
+  `}
+
+  ${props => !props.width && `
+    min-width: ${UNIT * 33}px;
+  `}
+`;
+
+function ContextMenu({
+  areaRef,
+  children,
+  deleteBlockFile,
+  type,
+  useContextItem,
+}: ContextMenuProps) {
+  const [anchorPoint, setAnchorPoint] = useState({ x: 0, y: 0 });
+  const [contextItem, setContextItem] = useState({} as ContextItemType);
+  const [visible, setVisible] = useState(false);
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+    document.addEventListener('contextmenu', handleContextMenu);
+    return () => {
+      document.addEventListener('click', handleClick);
+      document.removeEventListener('contextmenu', handleContextMenu);
+    };
+  });
```

Comment:
Could this use an empty dependency array for clarity?

Reply:
ESLint complains that it should include the two callback functions, which seems wrong, and suggests that I should remove the dep array if that isn't intended. Should I still add the empty array?

Reply:
yeah, u shud still have a dep array